packages/cli/src/build/helpers/overrides.ts

```diff
@@ -35,7 +35,7 @@ const getOverrideFilenames = ({
     case 'solid':
     case 'qwik': {
       // strip 'tsx', 'ts', 'jsx', 'js' from filename
-      const filenameStrippedFromExtensions = filename.replace(/(.jsx?|.tsx?)/, '');
+      const filenameStrippedFromExtensions = filename.replace(/(\.jsx?|\.tsx?)/, '');
```

Comment:
Should also add a `$` to make sure we're matching the end of the string:

```suggestion
      const filenameStrippedFromExtensions = filename.replace(/(\.jsx?|\.tsx?)$/, '');
```